### ISSUE TITLE
I added tab completions for fish shell

### ISSUE THREAD

**Issue by bagohart** [chmln/sd#166]
Hi, cool project!
I added tab completions for fish shell:
```
complete sd -x -s f -l flags -d "<flags> Regex flags. May be combined (like `-f mc`)."
complete sd -x -s h -l help -d "Prints help information"
complete sd -s s -l string-mode -d "Treat expressions as non-regex strings"
complete sd -s p -l preview -d "Output result into stdout and do not modify files"
complete sd -x -s V -l version -d "Prints version information"
```

Do you want to bundle them with the project, or would you like me to create a PR to have them included in the fish shell repo instead?

**Comment by CosmicHorrorDev** [comment]
Shell completions are automatically generated when building the executable. The output path is something like `target/release/build/sd-<some_hash>/out`. For `fish` it looks like this generates

```fish
complete -c sd -n "__fish_use_subcommand" -s n -d 'Limit the number of replacements'
complete -c sd -n "__fish_use_subcommand" -s f -l flags -d 'Regex flags. May be combined (like `-f mc`).'
complete -c sd -n "__fish_use_subcommand" -s p -l preview -d 'Output result into stdout and do not modify files'
complete -c sd -n "__fish_use_subcommand" -s s -l string-mode -d 'Treat expressions as non-regex strings'
complete -c sd -n "__fish_use_subcommand" -s r -d 'Recursively replace files'
complete -c sd -n "__fish_use_subcommand" -s h -l help -d 'Prints help information'
complete -c sd -n "__fish_use_subcommand" -s V -l version -d 'Prints version information'
```

I'm not too familiar with fish, so I don't know about the quality of the above complections (although I'm tempted to make the jump to either fish or nushell). It looks like this isn't included in the most recent release artifacts, but I plan to fix that before the next release

**Comment by alerque** [comment]
If for some reason the auto generated completions are not correct the crate that generates them should be fixed. Adding ad-hoc ones manually is not sustainable, they are bound to get out of sync the first time a non-fish user contributes a change to the CLI.

**Comment by CosmicHorrorDev** [comment]
I agree. I was asking to see if the manually generated completions were fixing an issue with the automatically generated one or if the author wasn't aware of the automatically generated completions because they weren't included with the most recent release

**Comment by bagohart** [comment]
Good question, it's been almost a year so tbh I'm not quite sure anymore.
I think that I had not been aware of the generated completions, although I searched the repo (using github search) before writing my own.

About the generated completions: The `-c` is optional, so it doesn't hurt to add it. I don't know the purpose of the `__fish_use_subcommand`
https://github.com/fish-shell/fish-shell/blob/master/share/functions/__fish_use_subcommand.fish

Fish-shell policy is to add manual completions if the man-page-generated ones are not as good as hand-written ones.
In this case, the generated ones are basically identically to what I wrote by hand, so I guess I'll close this issue.
